react/components/Slider.js

```diff
@@ -114,40 +112,30 @@ class Slider extends Component {
     this.init()
   }
 
-  enableTransition = (extraStyles = {}) => {
-    const { easing, duration } = this.props
-
-    setStyle(this._sliderFrame.current, {
-      ...getStylingTransition(easing, duration),
-      ...extraStyles
-    })
-  }
-
-  disableTransition = (extraStyles = {}) => {
-    const { easing } = this.props
-    setStyle(this._sliderFrame.current, {
-      ...getStylingTransition(easing),
-      ...extraStyles
-    })
-  }
-
   init = () => {
-    const { draggable, currentSlide, loop, cursor } = this.props
+    const { draggable, currentSlide } = this.props
     this.setSelectorWidth()
     this.setInnerElements()
     this.perPage = resolveSlidesNumber(this.props.perPage)
 
-    this.enableTransition({
-      width: `${(this.selectorWidth / this.perPage) * this.innerElements.length}px`,
-      ...(draggable ? { cursor } : {}),
+    requestAnimationFrame(() => {
+      const { easing, duration, cursor } = this.props
+      setStyle(this._sliderFrame.current, {
+        ...getStylingTransition(easing, duration),
+        width: `${this.totalSlides / this.perPage * 100}%`,
+        ...(draggable ? { cursor } : {}),  
+      })
+      requestAnimationFrame(() => {
```

Comment:
this `requestAnimationFrame` is okay I guess, but I think this could be improved by `setStyle` having a callback. but this is just a suggestion

Reply:
perhaps it's not even needed, because the upside of `getBoundingClientRect` triggering a re-layout is that you don't need to actually wait for it to be rendered, you could just call it right after setting the element's style.

but I'm not 100% sure its gonna work, so if you want to take heed of this suggestion you should test it